tests/model_validation/tests.py

```diff
@@ -29,28 +22,77 @@ def test_models_validate(self):
         #       See: https://code.djangoproject.com/ticket/21375
         management.call_command("check", stdout=six.StringIO())
 
-    def test_model_signal(self):
-        unresolved_references = post_init.unresolved_references.copy()
-        post_init.connect(on_post_init, sender='missing-app.Model')
-        post_init.connect(OnPostInit(), sender='missing-app.Model')
+    @isolate_apps('django.contrib.auth', kwarg_name='apps')
+    def test_lazy_reference_checks(self, apps):
+
+        class DummyModel(models.Model):
+            author = models.ForeignKey('Author', models.CASCADE)
+
+            class Meta:
+                app_label = "model_validation"
 
-        errors = run_checks()
+        class DummyClass(object):
+            def __call__(self, **kwargs):
+                pass
+
+            def dummy_method(self):
+                pass
+
+        def dummy_function(*args, **kwargs):
+            pass
+
+        apps.lazy_model_operation(dummy_function, ('auth', 'imaginarymodel'))
+        apps.lazy_model_operation(dummy_function, ('fanciful_app', 'imaginarymodel'))
+
+        post_init.connect(dummy_function, sender='missing-app.Model', apps=apps)
```

Comment:
Is it important to disconnect these listeners similar to what's done in tests/signals?